components/controlled_dialog.tsx

```diff
@@ -92,6 +94,18 @@ export class ControlledDialog extends React.Component<IControlledDialogProps, IS
               }}
               text={cancelButtonText || "Cancel"}
             />
+            {additionalButtons?.map(buttonProps => (
+              <Button
+                {...buttonProps}
+                className={`${Classes.POPOVER_DISMISS} ${buttonProps?.className || ""}`}
+                onClick={(e: React.MouseEvent<HTMLElement, MouseEvent>) => {
+                  if (!!buttonProps.onClick) {
+                    buttonProps.onClick(e);
+                  }
+                  this.setState({ isOpen: false });
```

Comment:
Mmm, I feel like this is going to cause trouble in the future. This whole component is getting too generic, trying to do too much. Might be easier to just roll your own dialog for the thing you want this for.

If we do keep this approach, you might want the buttonProps to control whether `isOpen` gets set to false on button click.

Reply:
I see, is looking a bit complex; will update